Sources/Amplitude/Events/BaseEvent.swift

```diff
@@ -269,4 +269,18 @@ extension BaseEvent {
         let decoder = JSONDecoder()
         return try? decoder.decode(T.self, from: jsonData)
     }
+
+    static func toJSONString(events: [BaseEvent]) -> String {
+        var result = ""
+        do {
+            let encoder = JSONEncoder()
+            let json = try encoder.encode(events)
+            if let printed = String(data: json, encoding: .utf8) {
+                result = printed
+            }
+        } catch {
+            result = ""
```

Comment:
Why catch the exception here? If the encoding fails, shouldn't this be sent as an error log too?

Reply:
There isn't any logger or diagonotiscs object when I put it here. I put this as a helper method and this was used only after we verified events desrialized, so didn't worry much on failure though as this was already verified once. If preferred, I could also put this to storage class then we could add error log there.

Reply:
I agree with Izaaz we should try to send this to diagnostics. I would vote to throw here and catch in Storage.

Reply:
Moved this to storage and added diagnostics there